nemo_curator/utils/semdedup_utils.py

```diff
@@ -179,22 +179,62 @@ def rank_within_cluster(
     return len(cluster_ids) - missing_files
 
 
-def _semdedup(
-    cluster_reps: torch.Tensor, device: str
+def pairwise_cosine_similarity(
+    cluster_reps: torch.Tensor,
+    device: Literal["cuda", "cpu"],
 ) -> Tuple[torch.Tensor, List[int]]:
-    # compute pairwise cos sim between cluster items,
-    # then replace to diagonal with zeros to ignore self similarity
-    cluster_reps.to(device)
-    pair_w_sim_matrix = cluster_reps @ (cluster_reps.T)
+    """
+    Compute pairwise cosine similarity between cluster items,
+    then replace to diagonal with zeros to ignore self similarity
+    """
+    # Move to device
+    cluster_reps = cluster_reps.to(device)
+    # Normalize embeddings
+    cluster_reps = cluster_reps / cluster_reps.norm(dim=1, keepdim=True)
+    # Compute pairwise cosine similarity
+    pairwise_sim_matrix = cluster_reps @ (cluster_reps.T)
```

Comment:
Haha I don't think I've used `@` before. Would there be any advantage to using `torch.mm`, `torch.matmul`, etc.?